app_build_suite/build_steps/helm.py

```diff
@@ -576,6 +654,10 @@ def cleanup(
             logger.info(f"Restoring backup {_chart_yaml}.back to {_chart_yaml}")
             chart_yaml_path = os.path.join(config.chart_dir, _chart_yaml)
             shutil.move(chart_yaml_path + ".back", chart_yaml_path)
+        if context_key_chart_lock_changes_made in context and context[context_key_chart_lock_changes_made]:
```

Comment:
You take this backup in `HelmRequirementsUpdater` but restore it here, in `HelmChartBuilder`. That's not a good idea to mix a single logical process across many classes. I think we should keep single owner for that.

Reply:
This is actually in the `HelmYAMLRestorer` step.

![image](https://user-images.githubusercontent.com/1494211/106625801-523a5400-6577-11eb-96d2-20aec0a2f8ad.png)

Reply:
Sorry, my bad. I had no idea why I thought it's in `ChartBuilder`